src/communication/index.ts

```diff
@@ -1,108 +1,69 @@
-import {
-  InterfaceMessageIds,
-  InterfaceMessageEvent,
-  SentSDKMessage,
-  SDKMessageIds,
-  SDKMessageToPayload,
-  RequestId,
-  InterfaceMessageToPayload,
-  Communicator,
-} from '../types';
-import { INTERFACE_MESSAGES } from './messageIds';
+import semver from 'semver';
+import { InterfaceMessageEvent, Communicator, Methods, Response } from '../types';
+import { MessageFormatter } from './messageFormatter';
 
-class InterfaceCommunicator implements Communicator {
-  private allowedOrigins: RegExp[] = [];
+// eslint-disable-next-line
+type Callback = (response: any) => void;
 
-  constructor(allowedOrigins: RegExp[]) {
+class PostMessageCommunicator implements Communicator {
+  private readonly allowedOrigins: RegExp[] | null = null;
+  private callbacks = new Map<string, Callback>();
+
+  constructor(allowedOrigins: RegExp[] | null = null) {
     this.allowedOrigins = allowedOrigins;
 
     window.addEventListener('message', this.onParentMessage);
   }
 
-  private isValidMessage({ origin, data }: InterfaceMessageEvent): boolean {
-    const emptyOrMalformed = !data || !data.messageId;
-    const unknownOrigin = this.allowedOrigins?.find((regExp) => regExp.test(origin)) === undefined;
-    const sameOrigin = origin === window.origin;
+  private isValidMessage = ({ origin, data, source }: InterfaceMessageEvent): boolean => {
+    const emptyOrMalformed = !data;
+    const sentFromParentEl = source === window.parent;
+    const allowedSDKVersion = typeof data.version !== 'undefined' ? semver.gte(data.version, '1.0.0') : false;
+    let validOrigin = true;
+    if (Array.isArray(this.allowedOrigins)) {
+      validOrigin = this.allowedOrigins.find((regExp) => regExp.test(origin)) === undefined;
```

Comment:
I am confused about this:

`find` return the item matching. So if an url in the `allowedOrigin` then it will be returned and `validOrigin` will evalute to `false` ... therefore if your url is in `allowedOrigin` it is not a `validOrigin` ... this feels weird

Reply:
I think I got it now, https://github.com/gnosis/safe-apps-sdk/pull/63/commits/e881992f0678e0ab922e40b6d6f0fc41ebda13fb should fix it